File: src/backend/src/main/java/com/personal/finance/backend/wallets/service/WalletService.java
package com.personal.finance.backend.wallets.service;

import com.personal.finance.backend.wallets.dto.request.CreateWalletRequest;
import com.personal.finance.backend.users.dto.request.UpdateWalletRequest;
import com.personal.finance.backend.wallets.dto.response.WalletDTO;
import com.personal.finance.backend.wallets.dto.response.WalletMemberDTO;

import java.util.List;

public interface WalletService {
    WalletDTO createWallet(Long ownerId, CreateWalletRequest request);
    List<WalletDTO> getWalletsForUser(Long userId);
    WalletDTO updateWallet(Long walletId, Long requesterId, UpdateWalletRequest request);
    void deleteWallet(Long walletId, Long requesterId);
    WalletMemberDTO shareWallet(Long walletId, Long requesterId, com.personal.finance.backend.wallets.dto.request.AddWalletMemberRequest request);
    List<WalletMemberDTO> getWalletMembers(Long walletId, Long requesterId);
}

File: src/frontend/src/services/walletService.ts
import api from './api';

const walletService = {
  getMyWallets: async () => {
    const response = await api.get('/wallets');
    return response.data; // Trả về List<WalletDTO>
  },

  createWallet: async (data: any) => {
    const response = await api.post('/wallets', data);
    return response.data;
  },

  updateWallet: async (id: number, data: any) => {
    const response = await api.put(`/wallets/${id}`, data);
    return response.data;
  },

  deleteWallet: async (id: number) => {
    const response = await api.delete(`/wallets/${id}`);
    return response.data;
  },

  getWalletMembers: async (walletId: number) => {
    const response = await api.get(`/wallets/${walletId}/members`);
    return response.data;
  },

  shareWallet: async (walletId: number, data: { email: string; permission: string }) => {
    const response = await api.post(`/wallets/${walletId}/members/share`, data);
    return response.data;
  }
};

export default walletService;